nextcord/http.py

```diff
@@ -240,151 +712,280 @@ async def request(
+        for retry_count in range(max_retry_count):  # To prevent infinite loops.
+            should_retry = False
+            try:
+                async with global_rate_limit, url_rate_limit:
+                    # This check is for asyncio.gather()'d requests where the rate limit can change.
+                    if (
+                        temp := self._get_url_rate_limit(route.method, route, auth)
+                    ) is not url_rate_limit and temp is not None:
+                        _log.debug(
+                            "Route %s had the rate limit changed, resetting and retrying.",
+                            rate_limit_path,
+                        )
+                        url_rate_limit = temp
+                        continue
+
+                    if files:
+                        for f in files:
+                            f.reset(seek=retry_count)
+
+                    if form:
+                        form_data = aiohttp.FormData(quote_fields=False)
+                        for params in form:
+                            form_data.add_field(**params)
+                        kwargs["data"] = form_data
+
+                    async with self.__session.request(
+                        method=route.method,
+                        url=route.url,
+                        headers=headers,
+                        proxy=self._proxy,
+                        proxy_auth=self._proxy_auth,
+                        **kwargs,
+                    ) as response:
                         _log.debug(
                             "%s %s with %s has returned %s",
-                            method,
-                            url,
+                            route.method,
+                            route.url,
                             kwargs.get("data"),
                             response.status,
                         )
 
-                        # even errors have text involved in them so this is safe to call
-                        data = await json_or_text(response)
-
-                        # check if we have rate limit header information
-                        is_global = bool(response.headers.get("X-RateLimit-Global", False))
-
-                        limit = response.headers.get("X-RateLimit-Limit", "")
-                        remaining = response.headers.get("X-Ratelimit-Remaining", "")
+                        await global_rate_limit.update(response)
+                        try:
+                            await url_rate_limit.update(response)
+                        except IncorrectBucket:
+                            # This condition can be met when doing asyncio.gather()'d requests.
+                            if (
+                                temp := self._buckets.get(
+                                    # Defaulting to "" makes pyright happy because None is an invalid type of key.
+                                    response.headers.get("X-RateLimit-Bucket", "")
```

Comment:
cast to str or Any

Reply:
I'm not sure how to do that?

Reply:
```suggestion
                                    cast(Any, response.headers.get("X-RateLimit-Bucket"))
```

make sure typing.Any and typing.Cast have been imported

Reply:
What's the purpose of casting to Any? I think the current solution works fine, since it just defaults to an empty string instead of trying to trick the type checker.